File: src/Domain/Library/Gap.cs
using SharedKernel;

namespace Domain.Library;

public sealed class Gap : Entity
{
    public Gap()
    {
        Manuscripts = new List<Manuscript>();
    }

    public int Id { get; set; }
    public bool State { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Note { get; set; } = string.Empty;

    public ICollection<Manuscript> Manuscripts { get; set; }
}

File: src/Web.UI/src/app/models/gap.ts
export type Gap = {
  id: number;
  title: string;
  note: string;
  state: boolean;
};

export type GapName = {
  id: number;
  title: string;
};

export type CreateGapRequest = {
  title: string;
  note: string;
  state: boolean;
};

export type UpdateGapRequest = CreateGapRequest & {
  id: number;
};


